src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -222,6 +223,40 @@ export class TabsTitleControl extends TitleControl {
 		});
 	}
 
+	private updateTabSizing(): void {
+		if (this.tabsContainer) {
+			const options = this.accessor.partOptions;
+			if (options.tabSizing === 'fixed') {
+				this.tabsContainer.style.setProperty('--tab-sizing-max-width', options.tabSizingMaxWidth + 'px');
+				this.tabsContainer.onmouseenter = () => this.fixTabWidths();
+				this.tabsContainer.onmouseleave = () => this.relaxTabWidths();
+			} else {
+				this.tabsContainer.style.removeProperty('--tab-sizing-max-width');
+				this.tabsContainer.onmouseenter = null;
+				this.tabsContainer.onmouseleave = null;
+				this.relaxTabWidths();
+			}
+		}
+	}
+
+	private fixTabWidths() {
+		if (this.accessor.partOptions.tabSizing === 'fixed') {
```

Comment:
I think you do not have to check on the option again, you do this already when calling.